etc/picongpu/summit-ornl/gpu_batch.tpl

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+# Copyright 2019 Axel Huebl, Rene Widera
+#
+# This file is part of PIConGPU.
+#
+# PIConGPU is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# PIConGPU is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with PIConGPU.
+# If not, see <http://www.gnu.org/licenses/>.
+#
+
+
+# PIConGPU batch script for Summit's LSF (bsub) batch system
+#   https://www.olcf.ornl.gov/for-users/system-user-guides/summit/summit-user-guide/#running-jobs
+
+#BSUB -q !TBG_queue
+#BSUB -W !TBG_wallTimeNoSeconds
+# Sets batch job's name
+#BSUB -J !TBG_jobName
+#BSUB -nnodes !TBG_nodes
+#BSUB -alloc_flags smt4
+# send me mails on job (-B)egin, Fi(-N)ish
+#BSUB !TBG_mailSettings -u !TBG_mailAddress
+#BSUB -cwd !TBG_dstPath
+#BSUB -P !TBG_nameProject
+
+#BSUB -o stdout.%J
+#BSUB -e stderr.%J
+
+
+## calculations will be performed by tbg ##
+.TBG_queue="batch"
+
+# remove seconds from walltime
+.TBG_wallTimeNoSeconds=${TBG_wallTime::-3}
+
+# settings that can be controlled by environment variables before submit
+.TBG_mailSettings=${MY_MAILNOTIFY:-""}
+.TBG_mailAddress=${MY_MAIL:-"someone@example.com"}
+.TBG_author=${MY_NAME:+--author \"${MY_NAME}\"}
+.TBG_nameProject=${proj:-""}
+.TBG_profile=${PIC_PROFILE:-"${PROJWORK}/!TBG_nameProject/${USER}/picongpu.profile"}
+
+# number of available/hosted GPUs per node in the system
+.TBG_numHostedGPUPerNode=6
+
+# required GPUs per node for the current job
+.TBG_gpusPerNode=`if [ $TBG_tasks -gt $TBG_numHostedGPUPerNode ] ; then echo $TBG_numHostedGPUPerNode; else echo $TBG_tasks; fi`
+
+# number of cores to block per GPU - we got 2x22 HW CPU cores per node
+#   and we will be accounted those anyway
+.TBG_coresPerGPU=7
+
+# use ceil to caculate nodes
+.TBG_nodes="$((( TBG_tasks + TBG_gpusPerNode - 1 ) / TBG_gpusPerNode))"
+
+## end calculations ##
+
+echo 'Running program...'
+echo !TBG_jobName
+
+cd !TBG_dstPath
+echo -n "present working directory:"
+pwd
+
+
+source !TBG_profile 2>/dev/null
+if [ $? -ne 0 ] ; then
+  echo "Error: PIConGPU environment profile under \"!TBG_profile\" not found!"
+  exit 1
+fi
+
+mkdir simOutput 2> /dev/null
+cd simOutput
+
+#jsrun  -N 1 -n !TBG_nodes !TBG_dstPath/input/bin/cuda_memtest.sh
+
+#if [ $? -eq 0 ] ; then
+export OMP_NUM_THREADS=7
+jsrun --nrs !TBG_tasks --tasks_per_rs 1 --cpu_per_rs 7 --gpu_per_rs 1 --rs_per_host !TBG_gpusPerNode --latency_priority GPU-CPU --bind rs --smpiargs="-gpu" !TBG_dstPath/input/bin/picongpu !TBG_author !TBG_programParams | tee output
```

Comment:
cosmetic:
```suggestion
jsrun --nrs !TBG_tasks --tasks_per_rs 1 --cpu_per_rs !TBG_coresPerGPU --gpu_per_rs 1 --rs_per_host !TBG_gpusPerNode --latency_priority GPU-CPU --bind rs --smpiargs="-gpu" !TBG_dstPath/input/bin/picongpu !TBG_author !TBG_programParams | tee output
```

Reply:
necessary change: with the current startup script and executing e.g. `16` GPUs the startup fails with:
```
Error: nrs (16) should be evenly divided rs_per_host (6)
```

cc @arghyac007

Reply:
Correct, if you use the -r feature it needs to be divisible by the -n, uneven distribution is not allowed. Wrt, your last comment, if you want to say run 9 resource sets, and just run -n 9, you’re guaranteed that all of the first node will be used and only 3 from the second node will be used (which 3, on which socket also cannot be controlled). 

The -r is mostly used if for some reason you want to use the whole CPU cores (42) but not all of the GPUs on one node, and maybe just 3 GPUs, one way to move the rest of the computation to the next node will be by using the -r.

Reply:
So you say we can just omit the `-r|--rs_per_host`?

The problem I face: I *wan* to use 16 GPUs for a problem (`--nrs`) but that will keep some nodes *intentionally* underutilized.

Reply:
Yup. You don’t need the -r, I would just use -n 16 -g 1. 

Since the allocation calculation is done in node hours, you always get exclusive node access, no one else will be able to use those nodes, so even if you’re using 2 GPUs on that node you’re billed for the entire node (hr).

Reply:
Thanks, testing right now.

Yes, I am aware of the full-node allocation policy as on titan and it makes a lot of sense. I am just thinking about large-scale simulations where one might have constrains in number of devices by static load balancing and where using 8000/8004 allocated GPUs is easier than finding a perfect match of 6 as well as decomposition :) (Just to keep things flexible for users, they should still try to use allocated resources efficiently, of course.)